pkg/service/users/users_service.go

```diff
@@ -0,0 +1,11 @@
+package users_service
+
+import (
+	"pkg/service/pkg/data/request"
+	"pkg/service/pkg/data/response"
+)
+
+type UsersService interface {
+	SaveUserAction(req request.CreateUserActivityRequest) error
```

Comment:
can you explain the value of having interfaces for the service ?

Reply:
I see it as a better practice to make an interface in order to decouple the implementation details from the code that uses the service (in this case the controller).
This way adding a mock implementation for testing should be much easier